### DISCUSSION TITLE
Background gradients w/ opacity control

### DISCUSSION THREAD

**Discussion by AHBruns** [tailwindlabs/tailwindcss#2429] [Ideas]
I'm making this as a feature request because I believe the current behavior is intended. However, the current behavior is confusing. So, this could probably also be listed as a bug report.

Basically, if I use `bg-gradient-to-{direction}` with `bg-opacity-{#}`, I'd expect to get a gradient background with # opacity. However, at the moment `bg-opacity-{#}` does nothing. So, my request is that the API be ammended to either apply `bg-opacity-{#}` to background gradients, or introduce a new `bg-gradient-opacity-{#}` set of classes.

An alternative would be to add `from-opacity-{#}`, `via-opacity-{#}`, and `to-opacity-{#}` class sets, but this seems like it might be too much bloat for only a small amount of extra control.

**Comment by TrevorGoertzen** [comment]
Does anyone know if there is a way to do this in the meantime?

**Reply by sayinserdar** [reply]
> Does anyone know if there is a way to do this in the meantime?

It's a naive approach but you can add the same color with expected opacity and then instead of ` bg-gradient-to-b from-viceBlue to-viceFuschia` you can use `bg-gradient-to-b from-viceBlue50 to-viceFuschia50`

```
  extend: {
      colors: {
        viceBlue: '#41B6E6',
        viceFuschia: '#DB3EB1',
        viceBlue50: 'rgb(219 62 177 / 50%)',
        viceFuschia50: 'rgb(65 182 230 / 50%)',
      },
    }
```
